--- wine-1.1.17/dlls/wined3d/directx.c	2009-02-28 01:21:46.000000000 +0300
+++ wine-1.1.17/dlls/wined3d/directx.c.patched	2009-03-11 16:33:31.000000000 +0300
@@ -2326,10 +2326,6 @@
         case WINED3DFMT_DXT3:
         case WINED3DFMT_DXT4:
         case WINED3DFMT_DXT5:
-            if (GL_SUPPORT(EXT_TEXTURE_COMPRESSION_S3TC)) {
-                TRACE_(d3d_caps)("[OK]\n");
-                return TRUE;
-            }
             TRACE_(d3d_caps)("[FAILED]\n");
             return FALSE;
 
--- wine-1.1.17/dlls/wined3d/device.c	2009-02-28 01:21:46.000000000 +0300
+++ wine-1.1.17/dlls/wined3d/device.c.patched	2009-03-11 16:28:39.000000000 +0300
@@ -6161,8 +6161,7 @@
                 } if (destFormat != srcFormat) {
                     FIXME("Updating mixed format compressed texture is not curretly support\n");
                 } else {
-                    GL_EXTCALL(glCompressedTexImage2DARB(glDescription->target, glDescription->level,
-                            dst_format_desc->glInternal, srcWidth, srcHeight, 0, destSize, data));
+		    FIXME("S3TC support dropped\n");
                 }
             } else {
                 FIXME("Attempting to update a DXT compressed texture without hardware support\n");
