diff -urN msgpack-0.5.7-orig//src/msgpack/type/define.hpp msgpack-0.5.7/src/msgpack/type/define.hpp
--- msgpack-0.5.7-orig//src/msgpack/type/define.hpp	2011-08-08 22:24:02.000000000 +0400
+++ msgpack-0.5.7/src/msgpack/type/define.hpp	2012-05-30 19:32:18.371674107 +0400
@@ -50,7 +50,7 @@
 	template <typename Packer>
 	void msgpack_pack(Packer& pk) const
 	{
-		pk.pack_array(1);
+		pk.pack_array(0);
 	}
 	void msgpack_unpack(msgpack::object o)
 	{
diff -urN msgpack-0.5.7-orig//src/msgpack/type/define.hpp.erb msgpack-0.5.7/src/msgpack/type/define.hpp.erb
--- msgpack-0.5.7-orig//src/msgpack/type/define.hpp.erb	2011-02-23 17:49:58.000000000 +0300
+++ msgpack-0.5.7/src/msgpack/type/define.hpp.erb	2012-05-30 19:31:55.061387923 +0400
@@ -50,7 +50,7 @@
 	template <typename Packer>
 	void msgpack_pack(Packer& pk) const
 	{
-		pk.pack_array(1);
+		pk.pack_array(0);
 	}
 	void msgpack_unpack(msgpack::object o)
 	{
