diff -urN msgpack-0.5.7-orig//src/vrefbuffer.c msgpack-0.5.7/src/vrefbuffer.c
--- msgpack-0.5.7-orig//src/vrefbuffer.c	2011-02-23 17:49:58.000000000 +0300
+++ msgpack-0.5.7/src/vrefbuffer.c	2012-05-30 19:35:44.623005018 +0400
@@ -170,7 +170,7 @@
 
 
 	const size_t nused = vbuf->tail - vbuf->array;
-	if(to->tail + nused < vbuf->end) {
+	if(to->tail + nused > to->end) {
 		const size_t tosize = to->tail - to->array;
 		const size_t reqsize = nused + tosize;
 		size_t nnext = (to->end - to->array) * 2;
