--- boost-log.orig/boost-log/libs/log/build/Jamfile.v2	2012-09-20 19:04:51.911585675 +0400
+++ boost-log/boost-log/libs/log/build/Jamfile.v2	2012-09-20 19:12:37.811190133 +0400
@@ -37,11 +37,7 @@
         <toolset>gcc:<cxxflags>-fno-strict-aliasing  # avoids strict aliasing violations in other Boost components
         <toolset>gcc-mingw:<linkflags>-Wl,--enable-auto-import
         <toolset>gcc-cygwin:<linkflags>-Wl,--enable-auto-import
-        <library>/boost/date_time//boost_date_time
-        <library>/boost/filesystem//boost_filesystem
-        <library>/boost/system//boost_system
         <threading>single:<define>BOOST_LOG_NO_THREADS
-        <threading>multi:<library>/boost/thread//boost_thread
     ;
 
 DEPENDS event_log_backend.cpp : simple_event_log.mc ;
@@ -81,6 +77,13 @@
         ws2_32
     : ## requirements ##
         <define>BOOST_LOG_BUILDING_THE_LIB=1
+        <threading>single:<find-shared-library>boost_date_time-1_48
+        <threading>single:<find-shared-library>boost_filesystem-1_48
+        <threading>single:<find-shared-library>boost_system-1_48
+        <threading>multi:<find-shared-library>boost_date_time-mt-1_48
+        <threading>multi:<find-shared-library>boost_filesystem-mt-1_48
+        <threading>multi:<find-shared-library>boost_system-mt-1_48
+        <threading>multi:<find-shared-library>boost_thread-mt-1_48
         <logapi>winnt
     ;
     
@@ -90,6 +93,13 @@
       ## unix sources ##
     : ## requirements ##
         <define>BOOST_LOG_BUILDING_THE_LIB=1
+        <threading>single:<find-shared-library>boost_date_time-1_48
+        <threading>single:<find-shared-library>boost_filesystem-1_48
+        <threading>single:<find-shared-library>boost_system-1_48
+        <threading>multi:<find-shared-library>boost_date_time-mt-1_48
+        <threading>multi:<find-shared-library>boost_filesystem-mt-1_48
+        <threading>multi:<find-shared-library>boost_system-mt-1_48
+        <threading>multi:<find-shared-library>boost_thread-mt-1_48
         <logapi>unix
     ;
 
